[CI] DataStreamsClientYamlTestSuiteIT test {p0=data_stream/200_rollover_failure_store/Rolling over a failure store on a data stream without the failure store enabled should work} failing

**Build Scans:**
- [elasticsearch-intake #7626 / part1](https://gradle-enterprise.elastic.co/s/akthh65vmybmy)
- [elasticsearch-pull-request #23382 / part-1](https://gradle-enterprise.elastic.co/s/hd57r7otnvvim)
- [elasticsearch-periodic-platform-support #3140 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/t5m3uomckxbx2)
- [elasticsearch-periodic-platform-support #3140 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/ljkynwckzcr2y)
- [elasticsearch-intake #7559 / part1](https://gradle-enterprise.elastic.co/s/euvihtzlxghfc)
- [elasticsearch-pull-request #22794 / part-1](https://gradle-enterprise.elastic.co/s/h23bg3gkxo6yi)

**Reproduction Line:**
```
./gradlew ':modules:data-streams:yamlRestTest' --tests "org.elasticsearch.datastreams.DataStreamsClientYamlTestSuiteIT.test {p0=data_stream/200_rollover_failure_store/Rolling over a failure store on a data stream without the failure store enabled should work}" -Dtests.seed=C90957F275DB1936 -Dtests.locale=ar-OM -Dtests.timezone=America/Santiago -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Bp0%3Ddata_stream%2F200_rollover_failure_store%2FRolling%20over%20a%20failure%20store%20on%20a%20data%20stream%20without%20the%20failure%20store%20enabled%20should%20work%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.datastreams.DataStreamsClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [data_stream/200_rollover_failure_store:417]: got unexpected warning header [
	299 Elasticsearch-8.15.0-97a7b83efe8d1814edd83616b1475359770dfa31 "index template [my-other-template] has index patterns [other-data-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [my-other-template] will take precedence during new index creation"
]

```

**Mute Reasons:**
- [main] 6 failures in test test {p0=data_stream/200_rollover_failure_store/Rolling over a failure store on a data stream without the failure store enabled should work} (1.9% fail rate)
- [main] 1.9% failure rate in test test {p0=data_stream/200_rollover_failure_store/Rolling over a failure store on a data stream without the failure store enabled should work} (1.9% fail rate)
- [main] 1.5% failure rate in step part-1 (1.5% fail rate)
- [main] 1.5% failure rate in pipeline elasticsearch-pull-request (1.5% fail rate)

**Note:**
This issue was created using new, in-development muting automation. Please mind any issues.

--- a/modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_rollover_failure_store.yml
+++ b/modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/200_rollover_failure_store.yml
@@ -416,7 +416,7 @@ teardown:
 "Rolling over a failure store on a data stream without the failure store enabled should work":
   - do:
       allowed_warnings:
-        - "index template [my-other-template] has index patterns [data-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [my-template] will take precedence during new index creation"
+        - "index template [my-other-template] has index patterns [other-data-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [my-other-template] will take precedence during new index creation"
       indices.put_index_template:
         name: my-other-template
         body:

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -80,9 +80,6 @@ tests:
 - class: "org.elasticsearch.action.admin.indices.rollover.RolloverIT"
   issue: "https://github.com/elastic/elasticsearch/issues/110034"
   method: "testRolloverWithClosedWriteIndex"
-- class: org.elasticsearch.datastreams.DataStreamsClientYamlTestSuiteIT
-  method: test {p0=data_stream/200_rollover_failure_store/Rolling over a failure store on a data stream without the failure store enabled should work}
-  issue: https://github.com/elastic/elasticsearch/issues/110051
 - class: org.elasticsearch.xpack.transform.transforms.TransformIndexerTests
   method: testMaxPageSearchSizeIsResetToConfiguredValue
   issue: https://github.com/elastic/elasticsearch/issues/109844

